handler/proxy.go

```diff
@@ -86,6 +97,51 @@ func (e *ErrorWrapper) Write(p []byte) (n int, err error) {
 	e.l.Error(strings.Replace(string(p), "\n", "", 1))
 	return len(p), nil
 }
+
 func newErrorLogWrapper(logger logrus.FieldLogger) *log.Logger {
 	return log.New(&ErrorWrapper{logger}, "", log.Lshortfile)
 }
+
+func (p *Proxy) applyWebsockets(req *http.Request) error {
+	bodyContent, _, diags := p.context.PartialContent(config.Proxy{Remain: p.context}.Schema(true))
+	if diags.HasErrors() {
+		return diags
+	}
+
+	if wss := bodyContent.Blocks.OfType("websockets"); len(wss) == 1 {
```

Comment:
```suggestion
	wss := bodyContent.Blocks.OfType("websockets")
	if len(wss) != 1 {
		return nil
	}
```

reduce method complexity